apps/els_lsp/src/els_text_search.erl

```diff
@@ -0,0 +1,46 @@
+%%==============================================================================
+%% Text-based search
+%%==============================================================================
+-module(els_text_search).
+
+%%==============================================================================
+%% API
+%%==============================================================================
+-export([ find_candidate_uris/1 ]).
+
+%%==============================================================================
+%% Includes
+%%==============================================================================
+-include("els_lsp.hrl").
+
+%%==============================================================================
+%% API
+%%==============================================================================
+-spec find_candidate_uris({els_dt_references:poi_category(), any()}) -> [uri()].
+find_candidate_uris(Id) ->
+  Pattern = extract_pattern(Id),
+  els_dt_document:find_candidates(Pattern).
+
+%%==============================================================================
+%% Internal Functions
+%%==============================================================================
+-spec extract_pattern({els_dt_references:poi_category(), any()}) ->
+        atom() | binary().
+extract_pattern({function, {_M, F, _A}}) ->
+  F;
+extract_pattern({type, {_M, F, _A}}) ->
+  F;
+extract_pattern({macro, {Name, _Arity}}) ->
+  Name;
+extract_pattern({macro, Name}) ->
+  Name;
+extract_pattern({include, Id}) ->
+  include_id(Id);
+extract_pattern({include_lib, Id}) ->
+  include_id(Id);
```

Comment:
Hmm... It looks like in the `word` splitting we save full strings and then compare for equality. But here we only extract the rootname. 

I think this means that include matching would never actually match since we have indexed strings like `"foo.hrl"`, but we're looking here for `"foo"`. Similar for `include_lib` we'd index `"app/include/foo.hrl"`

Reply:
That is correct, need to fix.

Reply:
Actually, we don't even need to index strings, with the exceptions of includes, so I may apply the same transformation on indexing.

Reply:
Fixed.